components/camera/camera.go

```diff
@@ -128,7 +111,8 @@ type Camera interface {
 //	err = myCamera.Close(context.Background())
 //
 // [camera component docs]: https://docs.viam.com/components/camera/
-type VideoSource interface {
+type Camera interface {
```

Comment:
There's a lot of files changed here. Am I correct that these changes to VideoSource/Camera/StreamCamera are the only real changes and everything else is the refactor to appease the compiler?

Reply:
Kind of - in the process of deprecating Stream, different methods are being used in vision, and the stream server now takes context to refresh the streams - I can break up this pr into independent pieces and remove some of sean yu's renames to make it smaller, I do think this was the diff that made the most sense for his manual testing.

Reply:
The camera driver changes + teh new `StreamCamera` interface (ffmpeg, transformpipeline, webcam) are there to appease the compiler without refactoring those drivers intensely.

Reply:
`StreamCamera` -> `VideoSource` https://github.com/viamrobotics/rdk/pull/4691/commits/64532fc351383645e561e402152ab5f2a18a0d39